### PR TITLE
Removing auth requirement for XDSS, retiring feature flags (SCP-4163)

### PR DESCRIPTION
**Pull request by bistline**
This update removes the requirement for users to be signed in for cross-dataset search requests.  In addition, the `cross_dataset_search_[frontend|backend]` feature flags have both been retired.

MANUAL TESTING
1. Pull this branch and run migrations with `bin/rails db:migrate`
2. In the Rails console, validate that both XDSS feature flags are gone:
```
irb(main):003:0> FeatureFlag.where(name: /cross_dataset_search/).count
=> 0
```
3. Boot as normal, and confirm that facet filters include results from Azul without needing to sign in
4. Run a search for `species:Homo sapiens` and `disease:HIV infectious disease` and confirm the HCA projects `An Atlas of Immune Cell Exhaustion in HIV-Infected Individuals Revealed by Single-Cell Transcriptomics` and `Single-Cell RNA-Seq Reveals Transcriptional Heterogeneity in Latent and Reactivated HIV-infected Cells` are returned
5. (Optional) sign in and re-run the search, confirming that the HCA results are the same

This PR satisfies SCP-4163.

### CODE REVIEW

**Review by ehanna4 — APPROVED** [review]
LGTM - code changes make sense to me and ran the steps locally and things worked as expected

**Comment by bistline** [comment]
Build failure is known issue that was corrected in #1403

File: app/javascript/components/HomePageContent.js [hunk 1]
```diff
@@ -1,15 +1,8 @@
...
 
 /** Until refactor is done, this merely sets appropriate component */
 export default function HomePageContent() {
-  const flags = getFeatureFlagsWithDefaults()
-  if (flags.cross_dataset_search_frontend) {
-    return (<HomePageContentXds />)
-  } else {
-    return (<HomePageContentLegacy />)
-  }
+  return (<HomePageContentLegacy />)
```
Comment (eweitz) [inline app/javascript/components/HomePageContent.js:7]: It'd help to move inline`HomePageContentLegacy.js` into this `HomePageContent.js`, or at least ticket as much for next sprint prep and note it as a ticket (e.g. "TODO (SCP-4321): inline HomePageContentLegacy.js into module").

**Review by eweitz — APPROVED** [review]

File: app/javascript/components/HomePageContent.js [hunk 2]
```diff
@@ -1,15 +1,92 @@
...
+      <MessageModal/>
+      <ProviderStack>
+        <LinkableSearchTabs/>
+      </ProviderStack>
+    </ErrorBoundary>
+  )
+}
+
+/** Include Reach router */
+export function HomePageContentLegacy() {
```
Comment (eweitz) [inline app/javascript/components/HomePageContent.js:83]:
```suggestion
export function HomePageContent() {
```

File: app/javascript/components/HomePageContent.js [hunk 3]
```diff
@@ -1,15 +1,92 @@
...
 
 /** Until refactor is done, this merely sets appropriate component */
 export default function HomePageContent() {
-  const flags = getFeatureFlagsWithDefaults()
-  if (flags.cross_dataset_search_frontend) {
-    return (<HomePageContentXds />)
-  } else {
-    return (<HomePageContentLegacy />)
-  }
+  return (<HomePageContentLegacy />)
```
Comment (eweitz) [inline app/javascript/components/HomePageContent.js:91]:
```suggestion
  return (<HomePageContent />)
```

**Review by eweitz — APPROVED** [review]
I should've noted we'll want to change the component function name, too.

**Review by devonbush — APPROVED** [review]
Looks good -- can you tell me more about what the `Regexp.quote` addition does?

**Comment by bistline** [comment]
> Looks good -- can you tell me more about what the `Regexp.quote` addition does?

This is apparently a lingering bug we had that wasn't covered in tests because XDSS was not enabled by default.  We have test coverage for regex escaping on text-based search, but it did not cover the facet matching functionality that happens during an XDSS search.  Removing the XDSS feature flag exposed the issue, which caused a test failure in the SearchControllerTest `test 'should return search results using keywords'`.

**Pull request merged**